diff --git a/node_modules/expo-modules-autolinking/android/expo-gradle-plugin/expo-autolinking-plugin-shared/src/main/kotlin/expo/modules/plugin/AutolinkigCommandBuilder.kt b/node_modules/expo-modules-autolinking/android/expo-gradle-plugin/expo-autolinking-plugin-shared/src/main/kotlin/expo/modules/plugin/AutolinkigCommandBuilder.kt
index 2064ca4..f7fc3cc 100644
--- a/node_modules/expo-modules-autolinking/android/expo-gradle-plugin/expo-autolinking-plugin-shared/src/main/kotlin/expo/modules/plugin/AutolinkigCommandBuilder.kt
+++ b/node_modules/expo-modules-autolinking/android/expo-gradle-plugin/expo-autolinking-plugin-shared/src/main/kotlin/expo/modules/plugin/AutolinkigCommandBuilder.kt
@@ -8,7 +8,7 @@ class AutolinkingCommandBuilder {
    * Command for finding and running `expo-modules-autolinking`.
    */
   private val baseCommand = listOf(
-    "node",
+    System.getProperty("NODE_BINARY", "node"),
     "--no-warnings",
     "--eval",
     "require('expo/bin/autolinking')",
diff --git a/node_modules/expo-modules-autolinking/android/expo-gradle-plugin/expo-autolinking-settings-plugin/src/main/kotlin/expo/modules/plugin/ExpoAutolinkingSettingsExtension.kt b/node_modules/expo-modules-autolinking/android/expo-gradle-plugin/expo-autolinking-settings-plugin/src/main/kotlin/expo/modules/plugin/ExpoAutolinkingSettingsExtension.kt
index 4255d95..4f271b4 100644
--- a/node_modules/expo-modules-autolinking/android/expo-gradle-plugin/expo-autolinking-settings-plugin/src/main/kotlin/expo/modules/plugin/ExpoAutolinkingSettingsExtension.kt
+++ b/node_modules/expo-modules-autolinking/android/expo-gradle-plugin/expo-autolinking-settings-plugin/src/main/kotlin/expo/modules/plugin/ExpoAutolinkingSettingsExtension.kt
@@ -57,7 +57,7 @@ open class ExpoAutolinkingSettingsExtension(
     File(
       settings.providers.exec { env ->
         env.workingDir(projectRoot)
-        env.commandLine("node", "--print", "require.resolve('@react-native/gradle-plugin/package.json', { paths: [require.resolve('react-native/package.json')] })")
+        env.commandLine(System.getProperty("NODE_BINARY", "node"), "--print", "require.resolve('@react-native/gradle-plugin/package.json', { paths: [require.resolve('react-native/package.json')] })")
       }.standardOutput.asText.get().trim(),
     ).parentFile
   }
@@ -69,7 +69,7 @@ open class ExpoAutolinkingSettingsExtension(
     File(
       settings.providers.exec { env ->
         env.workingDir(projectRoot)
-        env.commandLine("node", "--print", "require.resolve('react-native/package.json')")
+        env.commandLine(System.getProperty("NODE_BINARY", "node"), "--print", "require.resolve('react-native/package.json')")
       }.standardOutput.asText.get().trim(),
     ).parentFile
   }
diff --git a/node_modules/expo-modules-autolinking/android/expo-gradle-plugin/expo-autolinking-settings-plugin/src/main/kotlin/expo/modules/plugin/ExpoAutolinkingSettingsPlugin.kt b/node_modules/expo-modules-autolinking/android/expo-gradle-plugin/expo-autolinking-settings-plugin/src/main/kotlin/expo/modules/plugin/ExpoAutolinkingSettingsPlugin.kt
index 99ccc9f..0d941bd 100644
--- a/node_modules/expo-modules-autolinking/android/expo-gradle-plugin/expo-autolinking-settings-plugin/src/main/kotlin/expo/modules/plugin/ExpoAutolinkingSettingsPlugin.kt
+++ b/node_modules/expo-modules-autolinking/android/expo-gradle-plugin/expo-autolinking-settings-plugin/src/main/kotlin/expo/modules/plugin/ExpoAutolinkingSettingsPlugin.kt
@@ -36,7 +36,7 @@ open class ExpoAutolinkingSettingsPlugin : Plugin<Settings> {
     val expoModulesAutolinkingPath =
       settings.providers.exec { env ->
         env.workingDir(settings.rootDir)
-        env.commandLine("node", "--print", "require.resolve('expo-modules-autolinking/package.json', { paths: [require.resolve('expo/package.json')] })")
+        env.commandLine(System.getProperty("NODE_BINARY", "node"), "--print", "require.resolve('expo-modules-autolinking/package.json', { paths: [require.resolve('expo/package.json')] })")
       }.standardOutput.asText.get().trim()
 
     val expoAutolinkingDir = File(expoModulesAutolinkingPath).parentFile
