diff --git a/node_modules/react-native-gesture-handler/android/build.gradle b/node_modules/react-native-gesture-handler/android/build.gradle
index ac149a0..d6a6c44 100644
--- a/node_modules/react-native-gesture-handler/android/build.gradle
+++ b/node_modules/react-native-gesture-handler/android/build.gradle
@@ -42,7 +42,7 @@ def resolveReactNativeDirectory() {
     def reactNativePackage = file(
         providers.exec {
             workingDir(rootDir)
-            commandLine("node", "--print", "require.resolve('react-native/package.json')")
+            commandLine(System.getProperty("NODE_BINARY", "node"), "--print", "require.resolve('react-native/package.json')")
         }.standardOutput.asText.get().trim()
     )
 
